src/main/java/edu/cornell/kfs/fp/document/validation/impl/TotalAmountNonZeroValidation.java

```diff
@@ -0,0 +1,39 @@
+package edu.cornell.kfs.fp.document.validation.impl;
+
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.document.AccountingDocument;
+import org.kuali.kfs.sys.document.validation.GenericValidation;
+import org.kuali.kfs.sys.document.validation.event.AttributedDocumentEvent;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+public class TotalAmountNonZeroValidation extends GenericValidation {
+    private AccountingDocument accountingDocumentForValidation;
+
+    public boolean validate(AttributedDocumentEvent event) {
+        KualiDecimal sourceTotalAmount = accountingDocumentForValidation.getSourceTotal();
+        KualiDecimal targetTotalAmount = accountingDocumentForValidation.getTargetTotal();
+
+        boolean ret = true;
+
+        if(sourceTotalAmount == null || !sourceTotalAmount.isNonZero()){
```

Comment:
Please use ObjectUtils to check for nulls.

Reply:
perhaps it would make more sense to use the isZero method?

Reply:
Agreed, as a best practice it is always better to avoid using the ! operator when feasible. In ruby there is an unless(<expression>) statement that is the opposite of an if condition, meaning that it will always execute the code within unless the condition is met.